posts/Yoann/2017-11-27-introducing-hibernate-search-jsr352-mass-indexing-job.adoc

```diff
@@ -0,0 +1,163 @@
+= Introducing the Hibernate Search JSR-352 mass indexing job
+Yoann Rodière
+:awestruct-tags: [ "Hibernate Search", "JSR-352" ]
+:awestruct-layout: blog-post
+---
+
+Hibernate Search 5.9 will feature integration with JSR-352 runtimes,
+providing a new implementation of mass indexing (indexing a high volume of entities)
+as a JSR-352 job.
+
+Today we released 5.9.0.Beta1 so you can try it out already!
+
+Why is it nice? How to use it? Let's find out!
+
+== What's JSR-352?
+
+"JSR 352: Batch Applications for the Java Platform" is a Java EE specification
+defining interfaces and mechanisms to execute batch processes ("jobs"),
+i.e. processes that involve applying the same operation over and over again to a large number of items.
+
+In JSR-352:
+
+* The job definition is provided as an XML file and a few Java classes, called batch artifacts.
+  In our case, the definition is provided by Hibernate Search.
+* The JSR-352 runtime avoids boilerplate code by managing
+  threading, transactions, the execution flow (calling methods on batch artifacts), ...
+* And all that's left is configuring the infrastructure
+  In our case that's your responsibility, as the Hibernate Search user.
+
+== How is it better than the existing Mass Indexer?
+
+Ever tried to reindex 100,000 entities with the existing mass indexer,
+only to experience some network or disk error at the 99,001th entity?
+Then you have to restart from the very beginning,
+even though the first 99,000 entities were indexed just fine...
+
+With the JSR-352 mass indexing job, the job will simply resume from the last checkpoint,
+probably somewhere around the 95,000th entity.
+
+There are other advantages (such as being able to pool resources between multiple, unrelated background jobs),
+but this one really is the reason we think JSR-352 makes sense.
+
+== How to use the JSR-352 mass indexing job?
+
+=== Setup
+
+Within Wildfly https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-configuration-deploy-on-wildfly[use our JBoss modules],
+configure Hibernate ORM and Hibernate Search. Also add a dependency to the `org.hibernate.search.jsr352` module.
```

Comment:
How do I add that dependency?

Reply:
The same you would do with any other dependency to a WildFly. I'm not explaining how to add a dependency to the POM, I don't see why I would explain how to add a dependency to a WildFly deployment...

I'll add "to your deployment" in this sentence, maybe that's what you meant.

Reply:
Adding dependencies to a POM is common knowledge, whereas how to add specific module dependencies to a WF deployment is known to 1% of people. Also does it need the export flag for instance?

> In order for your application to access the Hibernate JSR 352 batch, [add a dependency](https://docs.jboss.org/author/display/WFLY10/Class+Loading+in+WildFly#ClassLoadinginWildFly-JBossDeploymentStructureFile) to the `org.hibernate.search.jsr352` module to your deployment, e.g. by specifying the following in your application's _META-INF/MANIFEST.MF_ file:
> `Dependencies: org.hibernate.search.jsr352`

That way all the info needed is there and people interested have a reference to more details.